src/forms/StripePaymentForm/StripePaymentForm.js

```diff
@@ -74,9 +82,84 @@ const cardStyles = {
   },
 };
 
+const OneTimePaymentWithCardElement = props => {
+  const { cardClasses, formId, handleStripeElementRef, hasCardError, error, label, intl } = props;
+  const labelText =
+    label || intl.formatMessage({ id: 'StripePaymentForm.saveAfterOnetimePayment' });
+  return (
+    <React.Fragment>
+      <label className={css.paymentLabel} htmlFor={`${formId}-card`}>
+        <FormattedMessage id="StripePaymentForm.creditCardDetails" />
+      </label>
+      <div className={cardClasses} id={`${formId}-card`} ref={handleStripeElementRef} />
+      {hasCardError ? <span className={css.error}>{error}</span> : null}
+      <div className={css.saveForLaterUse}>
+        <FieldCheckbox
+          className={css.saveForLaterUseCheckbox}
+          textClassName={css.saveForLaterUseLabel}
+          id="saveAfterOnetimePayment"
+          name="saveAfterOnetimePayment"
+          label={labelText}
+          value="saveAfterOnetimePayment"
+          useSuccessColor
+        />
+        <span className={css.saveForLaterUseLegalInfo}>
+          <FormattedMessage id="StripePaymentForm.saveforLaterUseLegalInfo" />
+        </span>
+      </div>
+    </React.Fragment>
+  );
+};
+
+const PaymentMethodSelector = props => {
+  const {
+    cardClasses,
+    formId,
+    changePaymentMethod,
+    defaultPaymentMethod,
+    handleStripeElementRef,
+    hasCardError,
+    error,
+    paymentMethod,
+    intl,
+  } = props;
+  const last4Digits = defaultPaymentMethod.attributes.card.last4Digits;
```

Comment:
Is it possible that there is no `attributes` or `card` in `defaultPaymentMethod`? Should it be checked or is it handled somewhere else?

Reply:
Excellent question. I added `    const ensuredDefaultPaymentMethod = ensurePaymentMethodCard(defaultPaymentMethod);
` to StripePaymentForm component. (I.e. I didn't repeat that on these subcomponents.)